x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/utils.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HostInfo, HostMetadata } from '../../../../common/endpoint/types';
+
+export const isPolicyOutOfDate = (
```

Comment:
This should be a selector and done in a way that it can be memoized (using `createSelector()`)

Reply:
This needs to be used by the list as well on an individual item.  We can select the `policy_info` but we'll need to pass in the individual item when the table is generated.  As long as the logic is reusable.

Reply:
in the list table, it is not using the store, or selectors. trying to do that actually broke functionality on the list table.

Reply:
after thinking about this more, I think we're OK to leave this as-is.  Here's my reasoning.

1. In details, we select the `policyInfo` and `details` before passing it into the `EndpointDetails` component.  At this point, the `isPolicyOutOfDate()` function just becomes a normal logic function that just takes parameters.
See this: https://github.com/elastic/kibana/pull/83469/files#diff-898766190cab3dfec5deafc2c55a7728c8ca667ed29dfaaa3dfc62832b6f3157R106

2. Similarly for the Endpoint list, we've already selected the `listData` and the `isPolicyOutOfDate()` function is used as a logical function on the data to see if we should show the UI component.
https://github.com/elastic/kibana/pull/83469/files#diff-b278029c5e0754f0be2b5c7fb170b37dcd12b494226e82ca64ab26527a79bc80R339

@paul-tavares let me know if you have other thoughts, but to me, all of the data has already been fetched via selectors and is passed into the `isPolicyOutOfDate()` function.  I'm OK with this as-is.  If needed, we can open a follow up ticket on Monday when you're back with an alternative approach.